androidTest/java/com/owncloud/android/authentication/SAMLAuthenticatorActivityTest.java

```diff
@@ -0,0 +1,290 @@
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isEnabled;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static android.support.test.espresso.web.sugar.Web.onWebView;
+import static android.support.test.espresso.web.webdriver.DriverAtoms.findElement;
+import static android.support.test.espresso.web.webdriver.DriverAtoms.webClick;
+import static android.support.test.espresso.web.webdriver.DriverAtoms.webKeys;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertTrue;
+
+
+@RunWith(AndroidJUnit4.class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+@LargeTest
+public class SAMLAuthenticatorActivityTest {
+
+    public static final String EXTRA_ACTION = "ACTION";
+    public static final String EXTRA_ACCOUNT = "ACCOUNT";
+
+    private static final int WAIT_INITIAL = 1000;
+    private static final int WAIT_LOGIN = 5000;
+    private static final int WAIT_CONNECTION = 2500;
+    private static final int WAIT_CHANGE = 1000;
+
+    private static final String ERROR_MESSAGE = "Activity not finished";
+    private static final String RESULT_CODE = "mResultCode";
+    private static final String LOG_TAG = "LoginSuiteSAML";
+
+    private Context targetContext = null;
+    private String testServerURL = null;
+    private String testUser = null;
+    private String testPassword = null;
+    private String webViewUsernameId = null;
+    private String webViewPasswordId = null;
+    private String webViewSubmitXPath = null;
+
+    public enum ServerType {
+        /*
+         *  Server with trusted certificate
+         */
+        TRUSTED(1),
+
+        /*
+         * Sever with non-trusted certificate
+         */
+        NON_TRUSTED(2);
+
+        private final int status;
+
+        ServerType (int status) {
+            this.status = status;
+        }
+
+        public int getStatus() {
+            return status;
+        }
+
+        public static ServerType fromValue(int value) {
+            switch (value) {
+                case 1:
+                    return TRUSTED;
+                case 2:
+                    return NON_TRUSTED;
+            }
+            return null;
+        }
+
+    }
+
+    public ServerType servertype;
+
+    @Rule
+    public ActivityTestRule<AuthenticatorActivity> mActivityRule = new ActivityTestRule<AuthenticatorActivity>(
+            AuthenticatorActivity.class) {
+        @Override
+        protected Intent getActivityIntent() {
+
+            targetContext = InstrumentationRegistry.getInstrumentation()
+                    .getTargetContext();
+            Intent result = new Intent(targetContext, AuthenticatorActivity.class);
+            result.putExtra(EXTRA_ACTION, AuthenticatorActivity.ACTION_CREATE);
+            result.putExtra(EXTRA_ACCOUNT, "");
```

Comment:
It makes nothing, can be safely removed. That parameter is ignored by ```AuhtenticatorActivity```  when ```EXTRA_ACTION```  is ```ACTION_CREATE```